### ISSUE TITLE
Improve label position

### ISSUE THREAD

**Issue by mtennekes** [zonebuilders/zonebuilder#20]
Method:

* calculate centroids for each zone, using directions and distances
* if centroid is inside polygon, use it, otherwise use any point in polygon (which may happen after intersection with area.

**Comment by urschrei** [comment]
An alternative approach is to use [polylabel](https://github.com/urschrei/polylabel) which guarantees that a point will be found inside the polygon, and which will (generally speaking) find a higher-quality label position.

**Comment by Robinlovelace** [comment]
Good thinking Martijn, are you up for giving it a go?

**Comment by mtennekes** [comment]
Yes, but it will be after the summer holidays. Hopefully before the conference.

This may be useful: https://twitter.com/edzerpebesma/status/1357372616126959621
